File: shared/domain/models/geometry.go
package models

// Coordinate は地理座標を表す。
type Coordinate struct {
	Lat float64 `json:"lat"`
	Lng float64 `json:"lng"`
}

// GeoJSONPolygon はGeoJSON Polygonジオメトリを表す。
// coordinates は [リング][頂点][lng, lat] の3次元配列。
// 最初のリングが外周、以降は穴（ホール）を表す。
type GeoJSONPolygon struct {
	Type        string       `json:"type"`        // "Polygon"
	Coordinates [][][2]float64 `json:"coordinates"` // [ring][point][lng,lat]
}

File: pwa/src/domain/models/geometry.ts
// 参照実装: shared/domain/models/geometry.go

/** 地理座標。 */
export interface Coordinate {
  lat: number;
  lng: number;
}

/**
 * GeoJSON Polygon ジオメトリ。
 * coordinates は [リング][頂点][lng, lat] の3次元配列。
 * 最初のリングが外周、以降は穴（ホール）を表す。
 */
export interface GeoJSONPolygon {
  type: "Polygon";
  coordinates: [number, number][][];
}
